### PR TITLE
feat: add docx support

### PR DESCRIPTION
**Pull request by aptlin**
Adds support for .docx documents. 

To test, run `yarn test:single -t docx`.

Partially resolves https://github.com/hwchase17/langchainjs/issues/499 (`mammoth` does not support the legacy .doc format)

### CODE REVIEW

File: docs/docs/modules/indexes/document_loaders/examples/file_loaders/docx.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+```bash npm2yarn
+npm install mammoth
+```
+
+# Usage
+
+```typescript
+import { DocxLoader } from "langchain/document_loaders";
+
+const loader = new PDFLoader(
```
Comment (MaximeThoonsen) [inline docs/docs/modules/indexes/document_loaders/examples/file_loaders/docx.md:20]: hey @aptlin , it should be DocxLoader rigth?

Comment (aptlin) [reply docs/docs/modules/indexes/document_loaders/examples/file_loaders/docx.md]: Right! Fixed in [5c00d4f](https://github.com/hwchase17/langchainjs/pull/556/commits/5c00d4f3ed2d82fbdf6a679bdc827cc7d3a67ad6)

**Review by nfcampos — APPROVED** [review]
This is great, thank you!

**Review by MaximeThoonsen — APPROVED** [review]

**Comment by aptlin** [comment]
Thanks! Synched the branch with main to get rid of the yarn.lock conflict

**Pull request merged**